lib/msal-browser/docs/caching.md

```diff
@@ -1,3 +1,88 @@
-## Cache Storage
+# Caching
 
-We offer two methods of storage for Msal, `localStorage` and `sessionStorage`.  Our recommendation is to use `sessionStorage` because it is more secure in storing tokens that are acquired by your users, but `localStorage` will give you Single Sign On accross tabs and user sessions.  We encourage you to explore the options and make the best decision for your application.
+When MSAL acquires a token, it caches it for future usage. MSAL manages token lifetimes and refreshing for you. APIs like `acquireTokenSilent()` retrieves tokens from the cache for a given account.
+
+```javascript
+const msal = require('@azure/msal-browser');
+
+// Create the msal application object
+const pca = new msal.PublicClientApplication({
+    auth: {
+        clientId: "Enter_the_Application_Id_Here", // e.g. "b1b60dca-c49d-496e-9851-xxxxxxxxxxxx" (guid)
+        authority: "https://login.microsoftonline.com/Enter_the_Tenant_Info_Here", // e.g. "common" or your tenantId (guid),
+        redirectUri: "/"
+    },
+    cache: {
+       cacheLocation: "sessionStorage"
+    }
+});
+
+/**
+* login* and acquireToken* APIs return an account object containing "homeAccountId"
+* you should keep a record of this in your app and use it later on when calling acquireTokenSilent
+* For more, see: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/accounts.md
+*/
+const someUserHomeAccountId = "Enter_User_Home_Account_Id";
+
+const account = await pca.getAccountByHomeId(someUserHomeAccountId);
+
+const silentTokenRequest = {
+    account: account,
+    scopes: ["user.read"],
+};
+
+pca.acquireTokenSilent(silentTokenRequest)
+    .then((response) => {
+        // do something with response
+    }).catch((error) => {
+        // catch and handle errors
+    });
+```
+
+## Cache storage
+
+By default, MSAL stores the various authentication artifacts it obtains from the IdP in browser storage using the [Web Storage API](https://developer.mozilla.org/docs/Web/API/Web_Storage_API) supported by all modern browsers. Accordingly, MSAL offers two methods of storage:
+
+- `sessionStorage` (default):
+- `localStorage`
+
+`sessionStorage` is more secure, but `localStorage` allows application or application instances on the same domain to share cache. This enables single sign-on between tabs and user sessions.
```

Comment:
"more secure" is a bit of a misnomer. Instead, it should be noted that session storage is not shared between tabs and is cleared when the tab is closed. Probably best to have a table to compares localstorage/sessionstorage/memorystorage as to when the cache is cleared (persistent, on tab close, on page refresh/navigation, etc), whether it is shared between tabs, and whether redirect operations are supported (not supported with memory storage).

Reply:
Also worth calling out that temporary cache entries will always be stored in session storage or in memory, and that msal will fallback to memory storage if local/session storage are not available.

Reply:
revised and added a table